fv_be/app/models/dictionary.py

```diff
@@ -0,0 +1,154 @@
+            "translation": self.translation,
+        }
+
+
+class AlternateSpelling(BaseSiteContentModel):
+    """Model for alternate spellings associated to each dictionary entry."""
+
+    # from fv:alternate_spelling, fv-word:alternate_spellings, fv-phrase:alternate_spellings
+    text = models.CharField(max_length=200)
+    dictionary_entry = models.ForeignKey(
+        "DictionaryEntry", on_delete=models.CASCADE, related_name="alternate_spellings"
+    )
+
+    def __str__(self):
+        return self.text
+
+
+class Pronunciation(BaseSiteContentModel):
+    """Model for pronunciations associated to each dictionary entry."""
+
+    # from fv-word:pronunciation
+    text = models.CharField(max_length=200)
+    dictionary_entry = models.ForeignKey(
+        "DictionaryEntry", on_delete=models.CASCADE, related_name="pronunciations"
+    )
+
+    def __str__(self):
+        return self.text
+
+
+class DictionaryEntry(BaseControlledSiteContentModel):
+    """Model for dictionary entries"""
+
+    class TypeOfDictionaryEntry(models.TextChoices):
+        # Choices for Type
+        WORD = "WORD", _("Word")
+        PHRASE = "PHRASE", _("Phrase")
+
+    # Fields
+    # from dc:title, relatively more max_length due to phrases
+    # todo: max_length may be modified after doing some analysis on the length of current phrases
+    title = models.CharField(max_length=800)
+    type = models.CharField(
+        max_length=6,
+        choices=TypeOfDictionaryEntry.choices,
+        default=TypeOfDictionaryEntry.WORD,
+    )
+    # todo: Link to site model when available
+    # may be inherited from an abstract base class or mixin later
+    # site = models.ForeignKey(
+    #     'Site',
+    #     on_delete=models.CASCADE
+    # )
+    # from fv-word:categories, fv-phrase:phrase_books
+    categories = models.ForeignKey(
+        Category,
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="dictionary_entries",
+    )
+    # from fv:custom_order
+    custom_order = models.CharField(max_length=200, blank=True)
+    # from fv-word:available_in_games, fvaudience:games
+    exclude_from_games = models.BooleanField(default=False)
+    # from fvaudience:children fv:available_in_childrens_archive
+    # exclude_from_kids can be a shared mixin for dictionary_entries, songs, stories and media
+    exclude_from_kids = models.BooleanField(default=False)
+    batch_id = models.CharField(max_length=255, blank=True)
+    # from fv:related_assets, fv-word:related_phrases
+    related_dictionary_entries = models.ManyToManyField(
+        "self",
+        blank=True,
+        symmetrical=False,
+        through="DictionaryEntrySelfAssociation",
+        related_name="incoming_related_dictionary_entries",
+    )
+
+    class Meta:
+        verbose_name_plural = _("DictionaryEntries")
+
+    def __str__(self):
+        return self.title
+
+
+class DictionaryEntrySelfAssociation(models.Model):
```

Comment:
I think this name could maybe be simpler... Maybe something like RelatedDictionaryEntry or DictionaryEntryLink?